tests/push/azure_repos_pull_request_test.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import json
+import subprocess
+
+import pytest
+
+from all_repos.push import azure_repos_pull_request
+from testing.auto_namedtuple import auto_namedtuple
+from testing.git import init_repo
+
+
+@pytest.fixture
+def fake_azure_repo(tmpdir):
+    # hax: make the repo end with project/slug so it "looks" like an azure repo
+    src = tmpdir.join('project/slug')
+    init_repo(src)
+
+    dest = tmpdir.join('dest')
+    subprocess.check_call(('git', 'clone', src, dest))
+    subprocess.check_call((
+        'git', '-C', dest, 'checkout', 'origin/master', '-b', 'feature',
+    ))
+    subprocess.check_call((
+        'git', '-C', dest, 'commit', '--allow-empty',
+        '-m', 'This is a commit message\n\nHere is some more information!',
+    ))
+    settings = azure_repos_pull_request.Settings(
+        api_key='fake-token',
+        organization='fake-org',
+        project='fake-project',
+    )
+    return auto_namedtuple(src=src, dest=dest, settings=settings)
+
+
+def test_azure_repos_pull_request(mock_urlopen, fake_azure_repo):
+    resp = {
+        'repository':
+        {
+            'webUrl': 'https://dev.azure.com/fake-org/fake-project/_git/slug',
+        },
+        'pullRequestId': '101',
+    }
+    mock_urlopen.return_value.read.return_value = json.dumps(resp).encode()
```

Comment:
there's a less error-prone api for this: https://github.com/asottile/all-repos/blob/main/testing/mock_http.py